### ISSUE TITLE
Failing to fetch artifact metadata since 4.0.0

### ISSUE THREAD

**Issue by TheMrMilchmann** [actions/deploy-pages#285]
After updating to deploy-pages@v4 and upload-pages-artifact@v3, I am running into the following error:

```
Fetching artifact metadata for github-pages in run 7265993097
Error: Fetching artifact metadata failed. Is githubstatus.com reporting issues with API requests, Pages or Actions? Please re-run the deployment at a later time.
Error: HttpError: Resource not accessible by integration
    at /home/runner/work/_actions/actions/deploy-pages/v4/node_modules/@octokit/request/dist-node/index.js:124:1
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at getArtifactMetadata (/home/runner/work/_actions/actions/deploy-pages/v4/src/internal/api-client.js:10:1)
    at Deployment.create (/home/runner/work/_actions/actions/deploy-pages/v4/src/internal/deployment.js:66:1)
    at main (/home/runner/work/_actions/actions/deploy-pages/v4/src/index.js:30:1)
Error: Error: Failed to create deployment (status: 403) with build version 5d6a87698b916de4c981c3373b41af57cfb2983f. Ensure GITHUB_TOKEN has permission "pages: write".
```

> Is githubstatus.com reporting issues with API requests, Pages or Actions?

No.

> Please re-run the deployment at a later time.

I've tried running this multiple times throughout the day.

> Error: Failed to create deployment (status: 403) with build version 5d6a87698b916de4c981c3373b41af57cfb2983f. Ensure GITHUB_TOKEN has permission "pages: write".

Interestingly, I did not change anything about the token permissions. In fact, the `pages: write` permission is explicitly granted in the workflow:

```yaml
name: Deploy
on:
  push:
    branches:
      - "master"


jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4

      - uses: actions/setup-node@v4
        with:
          node-version: "18.x"
          cache: "yarn"

      - run: yarn install

      - run: yarn generate

      - name: Upload artifact
        uses: actions/upload-pages-artifact@v3
        with:
          path: ./dist

  deploy:
    needs: build
    runs-on: ubuntu-latest
    permissions:
      pages: write
      id-token: write
    environment:
      name: github-pages
      url: ${{ steps.deployment.outputs.page_url }}
    steps:
      - name: Deploy
        id: deployment
        uses: actions/deploy-pages@v4
```

(This might be related to #284.)

**Comment by luizrcs** [comment]
Same here, and it's driving me crazy. I explicitly checked the failed job logs and it says right at the beginning the `pages` write permission is given.

**Comment by JamesMGreene** [comment]
This is probably due to your workflows missing the newly required `actions: read` permission for `v4`.

https://github.com/actions/deploy-pages/pull/283/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R33

We're going to investigate removing the need for that new permission in an in-flight follow-up PR today, so please bear with us:
- #288 

But, as a short-term workaround, I believe adding the `actions: read` permission to your workflow/job will alleviate this blocker.

**Comment by luizrcs** [comment]
> This is probably due to your workflows missing the newly required `actions: read` permission for `v4`.
> 
> https://github.com/actions/deploy-pages/pull/283/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R33
> 
> We're going to investigate removing the need for that new permission in an in-flight follow-up PR today, so please bear with us:
> 
> * [Use the Actions artifacts client to avoid needing the `actions: read` permission #288](https://github.com/actions/deploy-pages/pull/288)
> 
> But, as a short-term workaround, I believe adding the `actions: read` permission to your workflow/job will alleviate this blocker.

That solved the first issue for me, which was the 403 error, but now I'm, getting a 400 `No artifact_url provided` like another issue mentioned.

**Comment by JamesMGreene** [comment]
Yup, we've identified an issue on the API internal side. A fix is in progress and should be rolled out in the next few hours. Thank you all for your reporting and patience, and our apologies for the temporary inconvenience! 🙇 

P.S. If anyone needs a short-term workaround, please revert to using the previous major versions of `actions/upload-pages-artifact` and `actions/deploy-pages` until we can get the fix shipped.

**Comment by JamesMGreene** [comment]
We believe this should be resolved as of ~4 hours ago, assuming you also include the `actions: read` permission (for now). Please give it another try, but do let us know if you still encounter friction. 🙇

**Comment by TheMrMilchmann** [comment]
Thanks for the update! I can confirm that this is working now after adding the `actions: read` permission. Should this issue be kept open for now because it's technically not resolved yet (but hopefully with #288)?

**Comment by JamesMGreene** [comment]
📣 The latest release, [`v4.0.1`](https://github.com/actions/deploy-pages/releases/tag/v4.0.1), has **removed** the need for the `actions: read` permission that `v4.0.0` introduced. 🪓
